src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AppendFileStreamStrategy.cs

```diff
@@ -0,0 +1,193 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Strategies
+{
+    // this type serves some basic functionality that is common for native OS File Stream Strategies
+    internal sealed class AppendFileStreamStrategy : FileStreamStrategy
+    {
+        // To write to the end of file:
+        // On Windows, we need to pass 0xFFFFFFFFFFFFFFFF as offset to writefile() to append to EOF.
+        // On Unix, if the file is opened with O_APPEND the offset is ignored and we use write anyway.
+        private const long WriteOffset = unchecked((long)ulong.MaxValue);
+        private readonly SafeFileHandle _fileHandle; // only ever null if ctor throws
+
+        internal AppendFileStreamStrategy(SafeFileHandle handle, FileAccess access, FileShare share)
+        {
+            Debug.Assert(handle.IsAppend);
+
+            handle.EnsureThreadPoolBindingInitialized();
+            _fileHandle = handle;
+        }
+
+        internal AppendFileStreamStrategy(string path, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize)
+        {
+            _fileHandle = SafeFileHandle.Open(Path.GetFullPath(path), mode, access, share, options, preallocationSize);
+        }
+
+        public override bool CanSeek => _fileHandle.CanSeek;
+
+        public override bool CanRead => false; // this is how FileMode.Append works
+
+        public override bool CanWrite => !_fileHandle.IsClosed;
+
+        public override long Length => RandomAccess.GetFileLength(_fileHandle);
+
+        public override long Position
+        {
+            get => Length;
```

Comment:
I don't understand this logic.  On both .NET Framework 4.8 and .NET 5, this:
```C#
using System;
using System.IO;

class Program
{
    static void Main()
    {
        const string Path = @"tmp.txt";
        File.Delete(Path);

        using (var fs = new FileStream(Path, FileMode.Append))
        {
            fs.Write(new byte[] { 1, 2, 3, 4, 5 }, 0, 5);
            fs.Position = 1;
            Console.WriteLine(fs.Position);
            Console.WriteLine(fs.Length);
        }
    }
}
```
prints:
```
1
5
```
With the logic you have here, won't this print:
```
5
5
```
?